components/shared/embedder/input_events.rs

```diff
@@ -280,3 +284,62 @@ pub enum GamepadUpdateType {
     /// <https://www.w3.org/TR/gamepad/#dfn-represents-a-standard-gamepad-button>
     Button(usize, f64),
 }
+
+pub trait WebDriverInputEvent {
+    fn webdriver_msg_id(&self) -> Option<WebDriverMessageId>;
+    fn with_webdriver_msg_id(&mut self, webdriver_id: Option<WebDriverMessageId>) -> Self;
+}
+
+impl WebDriverInputEvent for InputEvent {
+    fn webdriver_msg_id(&self) -> Option<WebDriverMessageId> {
```

Comment:
Nit: Let's have the method name match the type name:

```suggestion
    fn webdriver_message_id(&self) -> Option<WebDriverMessageId> {
```